definitions/scenarios/upgrade_to_katello_3_14.rb

```diff
@@ -0,0 +1,86 @@
+module Scenarios::Katello_3_14
+  class Abstract < ForemanMaintain::Scenario
+    def self.upgrade_metadata(&block)
+      metadata do
+        tags :upgrade_scenario
+        confine do
+          feature(:instance).upstream? && feature(:foreman_server) && \
+            feature(:katello) && (feature(:katello).current_version.major_minor == '3.13' || \
+            ForemanMaintain.upgrade_in_progress == '3.14')
+        end
+        instance_eval(&block)
+      end
+    end
+
+    def target_version
+      '3.14'
+    end
+  end
+
+  class PreUpgradeCheck < Abstract
+    upgrade_metadata do
+      description 'Checks before upgrading to Katello 3.14'
+      tags :pre_upgrade_checks
+      run_strategy :fail_slow
+    end
+
+    def compose
+      add_steps(find_checks(:default))
+      add_steps(find_checks(:pre_upgrade))
+      add_step(Checks::Repositories::Validate.new(:version => '3.14'))
+    end
+  end
+
+  class PreMigrations < Abstract
+    upgrade_metadata do
+      description 'Procedures before migrating to Katello 3.14'
+      tags :pre_migrations
+    end
+
+    def compose
+      add_steps(find_procedures(:pre_migrations))
+      add_step(Procedures::Service::Stop.new)
+    end
+  end
+
+  class Migrations < Abstract
+    upgrade_metadata do
+      description 'Migration scripts to Katello 3.14'
+      tags :migrations
+    end
+
+    def compose
+      add_step(Procedures::Repositories::Setup.new(:version => '3.14'))
+      add_step(Procedures::Packages::UnlockVersions.new)
+      add_step(Procedures::Packages::Update.new(:assumeyes => true))
+      add_step(Procedures::Installer::Upgrade.new)
```

Comment:
Here installer will execute database migrations then the step `Procedures::Foreman::VacuumDatabase` can be run as part of katello upgrade.
What do you think?

Reply:
What I think is vacuum should not be part of upgrade as it optional step, sometimes it take hours to complete. I am thinking to take user consent before performing vacuum in post upgrade steps instead of calling it without consent ?

Reply:
> What I think is vacuum should not be part of upgrade as it optional step

IIRC I have added this suggestion as I found an inconsistency where you included this step only inside foreman upgrade scenarios not inside Katello scenarios. I wanted to confirm the behavior.

> I am thinking to take user consent before performing vacuum in post upgrade steps instead of calling it without consent ?

It could be good idea or having an option for it while running sub-command may be helpful. 

My point was just that `Procedures::Foreman::VacuumDatabase` step will be useful after database migration so we can include it in Katello scenarios as well.